pkg/controller/perconaservermongodb/status.go

```diff
@@ -169,6 +169,10 @@ func (r *ReconcilePerconaServerMongoDB) updateStatus(cr *api.PerconaServerMongoD
 		cr.Status.State = api.AppStateError
 	}
 
+	if cr.Status.State != api.AppStateError && cr.Status.Mongos.Status != api.AppStateReady {
```

Comment:
cr.Status.Mongos can be nil
```suggestion
	if cr.Status.State != api.AppStateError && cr.Status.Mongos != nil && cr.Status.Mongos.Status != api.AppStateReady {
```

Reply:
Thanks!
https://github.com/percona/percona-server-mongodb-operator/pull/494/commits/77f8656f517c95d9973c69e83ffd4d5d9663d744